api/lib/source/polkadotV0-source.js

```diff
@@ -487,9 +487,30 @@ class polkadotAPI {
   async getUndelegationsForDelegatorAddress(address) {
     const api = await this.getAPI()
 
-    const stakingledger = await api.query.staking.ledger(address)
-    const undelegations = stakingledger.toJSON().unlocking
-    return undelegations.map((undelegation) =>
+    const [stakingLedger, progress] = await Promise.all([
+      api.query.staking.ledger(address),
+      api.derive.session.progress()
+    ])
+    const undelegations = stakingLedger.toJSON().unlocking
+    // each hour in both Kusama and Polkadot has 600 slots, one block per slot maximum
+    const eraBlocks = (24 * 600) / this.network.erasPerDay
```

Comment:
is it certain, that each slot is filled with a block?

Reply:
I did the calculation doing this assumption and it matched 100% the number of blocks according to Polkadot JS, so they must do it too

Reply:
probably doesn't matter anyway if it misses a block or two